scratch/code_generation_middleware.R

```diff
@@ -0,0 +1,68 @@
+library(tidyverse)
+library(rlang)
+devtools::load_all()
+
+parent <- tibble(a = c(1L, 1:3), b = -1)
+child <- tibble(a = 1:4, c = 3)
+
+dm <- dm(parent, child)
+
+enum_ops <- function(dm = NULL, ..., op_name = NULL) {
+  if (is.null(op_name)) {
+    list(
+      op_name = c(
+        "dm_add_pk", # only if not all tables have PK
+        "dm_add_fk",
+        NULL
+      )
+    )
+  } else {
+    list2(
+      op_name = op_name,
+      !!!exec(paste0("enum_ops_", op_name), dm = dm, ...)
+    )
```

Comment:
Just a minor question/comment (as I familiarize myself with the code)—Could this be simpler as as a direct function call?:
```R
c(op_name = op_name,
  get(paste0("enum_ops_", op_name))(dm = dm, ...))
```
(Ditto for `enum_ops_dm_add_pk()`.)

Reply:
Not sure, are there limitations with the current approach?

Reply:
> @egnha: Could you please take a look at what a first set of tests could look like for the existing functionality, even if it's not perfect?

Sure. As a new issue, first?